### ISSUE TITLE
[BUG] Ruby client says 2.4 is supported, but it errors

### ISSUE THREAD

**Issue by jason-o-matic** [OpenAPITools/openapi-generator#13986]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

https://github.com/OpenAPITools/openapi-generator/blob/b1c6513015531f83418d393092a04293d424ded9/modules/openapi-generator/src/main/resources/ruby-client/base_object.mustache#L16 uses the `transform_keys` method on Hash, which was [introduced in ruby 2.5]( https://www.bigbinary.com/blog/ruby-2-5-adds-hash-transform_keys-method ).

There are several places that reference versions <2.5:
- https://github.com/OpenAPITools/openapi-generator/blob/6229801935dc439e9b4b2c81bc12e4642333b038/modules/openapi-generator/src/main/resources/ruby-client/travis.mustache#L4
- https://github.com/OpenAPITools/openapi-generator/blob/b97a4ad12a59025fa51244d082f475315987a69a/modules/openapi-generator/src/main/resources/ruby-client/gitlab-ci.mustache#L10
- https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/ruby.md

Seems like the minimum version should be 2.5, or `transform_keys` should be replaced with something 2.3/2.4 compatible.

Here's the error:
```
/app/lib/docraptor/models/async_doc.rb:111:in `build_from_hash': undefined method `transform_keys' for {:status_id=>"[REDACTED]"}:Hash (NoMethodError)
Did you mean?  transform_values
	from /app/lib/docraptor/models/async_doc.rb:103:in `build_from_hash'
	from /app/lib/docraptor/api_client.rb:282:in `convert_to_type'
	from /app/lib/docraptor/api_client.rb:242:in `deserialize'
	from /app/lib/docraptor/api_client.rb:73:in `call_api'
	from /app/lib/docraptor/api/doc_api.rb:76:in `create_async_doc_with_http_info'
	from /app/lib/docraptor/api/doc_api.rb:27:in `create_async_doc'
	from async.rb:14:in `<main>'
```

##### openapi-generator version

Discovered on v6.1.0, but the above links reflect the same code in master.

##### OpenAPI declaration file content or url

https://github.com/DocRaptor/docraptor-ruby/blob/master/docraptor.yaml is an example that produces this issue.

##### Generation Details

```
generate \
  -i docraptor.yaml \
  -g ruby \
  -c "$CONFIG_FILE"
```
Here's the config: https://github.com/DocRaptor/docraptor-ruby/blob/master/swagger-config.json

##### Steps to reproduce

1. Generate client
2. Use ruby 2.4 to make an API request that creates a model object from the response

##### Related issues/PRs

I searched for `transform_keys` but nothing came up.

##### Suggest a fix

Given ruby EOL status it seems reasonable to drop support for <2.5. On the other hand replacing `transform_keys` seems like it'd be pretty easy if you wanted to preserve old version support.

**Comment by wing328** [comment]
thanks for reporting the issue. i'll pump the version to 2.7 instead, has not yet reached EOL: https://endoflife.date/ruby

**Comment by wing328** [comment]
UDPATE: closed via #14002
